.github/workflows/build-and-test.yml

```diff
@@ -26,6 +26,22 @@ jobs:
           - pypy-3.6
           - pypy-3.7
           - pypy-3.8
+        astroid-version:
+          - ''
+        include:
+          # Test recent Python on Astroid v2 too (the above natural tests will
+          # pick up Astroid v3 for these Python versions)
+          - python-version: '3.8'
+            astroid-version: '<3'
+          - python-version: '3.9'
+            astroid-version: '<3'
+          - python-version: '3.10'
+            astroid-version: '<3'
+          - python-version: '3.11'
+            astroid-version: '<3'
+          - python-version: '3.12'
+            astroid-version: '<3'
```

Comment:
This doesn't feel the clearest spelling of this, however given the non-overlapping version requirements between Python 2 & 3 it is probably the simplest.